src/foundations/rhi/Texture.zig

```diff
@@ -4,45 +4,68 @@ handle: c.GLuint64 = 0,
 uniform: Uniform = undefined,
 wrap_s: c.GLint = c.GL_CLAMP_TO_EDGE,
 wrap_t: c.GLint = c.GL_CLAMP_TO_EDGE,
+disable_bindless: bool = false,
 
 const Texture = @This();
 
 pub const TextureError = error{
     BindlessHandleCreationFailed,
-    BindlessNotSupported,
 };
 
-pub fn init(image: *assets.Image, program: u32, uniform_name: []const u8) TextureError!Texture {
-    var t: Texture = .{};
-    return t.setup(image, program, uniform_name);
+pub fn frag_shader(tx: ?Texture) Shader.fragment_shader_type {
+    if (tx) |t| if (!t.disable_bindless) return .bindless;
+    return .texture;
 }
 
-pub fn setup(self: *Texture, image: *assets.Image, program: u32, uniform_name: []const u8) TextureError!Texture {
+pub fn init(disable_bindless: bool) TextureError!Texture {
+    var t: Texture = .{
+        .disable_bindless = disable_bindless,
+    };
+
     if (c.glfwExtensionSupported("GL_ARB_bindless_texture") != 1 or c.glfwExtensionSupported("GL_ARB_bindless_texture") != 1) {
-        return TextureError.BindlessNotSupported;
+        t.disable_bindless = true;
     }
+    return t;
+}
 
+pub fn setup(self: *Texture, image: ?*assets.Image, program: u32, uniform_name: []const u8) TextureError!void {
     var name: u32 = undefined;
     c.glCreateTextures(c.GL_TEXTURE_2D, 1, @ptrCast(&name));
     c.glTextureParameteri(name, c.GL_TEXTURE_WRAP_S, self.wrap_s);
     c.glTextureParameteri(name, c.GL_TEXTURE_WRAP_T, self.wrap_t);
     c.glTextureParameteri(name, c.GL_TEXTURE_MIN_FILTER, c.GL_LINEAR_MIPMAP_LINEAR);
     c.glTextureParameteri(name, c.GL_TEXTURE_MAG_FILTER, c.GL_LINEAR);
-    const w: f32 = @floatFromInt(image.width);
-    const h: f32 = @floatFromInt(image.height);
-    const mip_map_levels: c.GLsizei = @intFromFloat(@ceil(@log2(@max(w, h))));
-    c.glTextureStorage2D(name, mip_map_levels, c.GL_RGBA8, @intCast(image.width), @intCast(image.height));
-    c.glTextureSubImage2D(
-        name,
-        0,
-        0,
-        0,
-        @intCast(image.width),
-        @intCast(image.height),
-        c.GL_RGBA,
-        c.GL_UNSIGNED_BYTE,
-        image.stb_data.ptr,
-    );
+    if (image) |img| {
```

Comment:
**suggestion:** Consider handling texture creation failure

The code creates a magenta texture when no image is provided, but doesn't handle the case where texture creation fails. Consider adding error handling for this scenario.